src/app/WriteHandler.cpp

```diff
@@ -50,20 +48,10 @@ void WriteHandler::Shutdown()
 {
     VerifyOrReturn(mState != State::Uninitialized);
     mMessageWriter.Reset();
-    ClearExistingExchangeContext();
```

Comment:
Should we add the below in SendWriteResponse function, after you remove ClearExistingExchangeContext?

   if (err != CHIP_NO_ERROR)
    {
        mpExchangeCtx->Close();
    }

Reply:
Also in https://github.com/project-chip/connectedhomeip/blob/master/src/app/CommandHandler.cpp#L75, it seems the exchange context close is also removed in command

Reply:
> Should we add the below in SendWriteResponse function, after you remove ClearExistingExchangeContext?

No, because `SendWriteResponse` is being called directly from `OnWriteRequest`, which is called from `InteractionModelEngine::OnMessageReceived`.  The exchange will close itself when we return from `OnMessageReceived`.  Note that we used to null out `mpExchangeCtx` on `OnWriteRequest` to prevent the `Shutdown` from closing the exchange there....

> it seems the exchange context close is also removed in command somehow, previously it has. When send response fails, we should close exchange context.

Again, that code is under `OnMessageReceived` and that handles the closing.  If we start responding async, then we will need to handle closing the exchange at that point.